product/roundhouse.tests/sqlsplitters/MySqlStatementSplitterSpecs.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using roundhouse.databases.mysql.parser;
+using roundhouse.sqlsplitters;
+
+namespace roundhouse.tests.sqlsplitters
+{
+    public class MySqlStatementSplitterSpecs
+    {
+        public abstract class concern_for_StatementSplitter : TinySpec
+        {
+            public override void Context()
+            {
+                
+            }
+        }
+
+        [Concern(typeof(StatementSplitter))]
+        public class when_splitting_sql_statements : concern_for_StatementSplitter
+        {
+            public override void Because()
+            {
+
+            }
+
+            [Observation]
+            public void should_split_on_delimiters()
+            {
+                string script = "select * from test1;\nselect * from test2;\n";
+
+                Parser parser = new Parser(script);
+                List<ParsedStatement> statements = parser.Parse();
+                WriteStatements(statements);
+
+                TestContext.Out.WriteLine("Statements parsed: " + statements.Count);
+                Assert.IsTrue(statements[0].Value.Length > 10);
```

Comment:
Nitpicking again, sorry. And, I do realise that the readability of the tests in RoundhousE is not great, but we always try to improve, don't we? ;), but I would prefer if you used the `Assert.Greater` instead, as the output if the test fails is much better: 

```suggestion
Assert.Greater(statements[0].Value.Length, 10);
```

Example, see if I change the test to test on 50 instead of 10. The `Assert.IsTrue` outputs:
```
roundhouse.tests.sqlsplitters.MySqlStatementSplitterSpecs+when_splitting_sql_statements.should_split_on_delimiters

  Expected: True
  But was:  False
```

However, if you use `Assert.Greater`, it outputs exactly what is the problem, so fixing the error is much easier:

```
roundhouse.tests.sqlsplitters.MySqlStatementSplitterSpecs+when_splitting_sql_statements.should_split_on_delimiters

  Expected: greater than 50
  But was:  20
```

This is _not_ meant as picking on you, definitely, just hoping to improve the quality of the tests of RoundhousE going forward.

Another thought, what does this assert really test? Yes, it tests that the first token is longer than 10, but it that the essence of the test? I see that testing that the count is 2 is definitely testing the logic, and makes sense. I just wonder what scenario the first assert is testing.

*Please*, if you feel this it tedious, and I'm being a paint in the butt, do tell me. It's well meaning, but if you feel it's a bit too much, just say so, and I'll be happy to merge the PR. It does add a lot of business value (i.e. fixing a bug) :)

Reply:
My concern is that it would do something weird, like only count the trailing delimiter and provide two statements: one with all of the test text and a second statement that was empty or only white space.

Another thing I was trying to test was the preservation of white space and comments. Most of the time these don't matter but I've seen people put a lot of comments in stored procedures.

I will clean these up a bit and try to make the things being tested a little more clear.

Thank you!